mls-rs/src/group/proposal_filter/filtering.rs

```diff
@@ -460,34 +517,58 @@ pub(crate) fn proposer_can_propose(
-                | ProposalType::REMOVE
-                | ProposalType::PSK
-                | ProposalType::RE_INIT
-                | ProposalType::GROUP_CONTEXT_EXTENSIONS
-        ),
-        (Sender::Member(_), ProposalSource::ByReference(_)) => matches!(
-            proposal_type,
-            ProposalType::ADD
-                | ProposalType::UPDATE
-                | ProposalType::REMOVE
-                | ProposalType::PSK
-                | ProposalType::RE_INIT
-                | ProposalType::GROUP_CONTEXT_EXTENSIONS
-        ),
+        (Sender::Member(_), ProposalSource::ByValue | ProposalSource::Local) => {
+            let can_propose = matches!(
+                proposal_type,
+                ProposalType::ADD
+                    | ProposalType::REMOVE
+                    | ProposalType::PSK
+                    | ProposalType::RE_INIT
+                    | ProposalType::GROUP_CONTEXT_EXTENSIONS
+            );
+            #[cfg(all(
+                feature = "by_ref_proposal",
+                feature = "custom_proposal",
+                feature = "self_remove_proposal"
+            ))]
+            let can_propose = can_propose || matches!(proposal_type, ProposalType::SELF_REMOVE);
+            can_propose
+        }
+        (Sender::Member(_), ProposalSource::ByReference(_)) => {
+            let can_propose = matches!(
+                proposal_type,
+                ProposalType::ADD
+                    | ProposalType::UPDATE
+                    | ProposalType::REMOVE
+                    | ProposalType::PSK
+                    | ProposalType::RE_INIT
+                    | ProposalType::GROUP_CONTEXT_EXTENSIONS
+            );
+            #[cfg(all(
+                feature = "by_ref_proposal",
+                feature = "custom_proposal",
+                feature = "self_remove_proposal"
+            ))]
+            let can_propose = can_propose || matches!(proposal_type, ProposalType::SELF_REMOVE);
+            can_propose
+        }
         #[cfg(feature = "by_ref_proposal")]
         (Sender::External(_), ProposalSource::ByValue) => false,
         #[cfg(feature = "by_ref_proposal")]
-        (Sender::External(_), _) => matches!(
-            proposal_type,
-            ProposalType::ADD
-                | ProposalType::REMOVE
-                | ProposalType::RE_INIT
-                | ProposalType::PSK
-                | ProposalType::GROUP_CONTEXT_EXTENSIONS
-        ),
+        (Sender::External(_), _) => {
+            let can_propose = matches!(
+                proposal_type,
+                ProposalType::ADD
+                    | ProposalType::REMOVE
+                    | ProposalType::RE_INIT
+                    | ProposalType::PSK
+                    | ProposalType::GROUP_CONTEXT_EXTENSIONS
+            );
+
+            #[cfg(all(feature = "custom_proposal", feature = "self_remove_proposal"))]
+            let can_propose = can_propose || matches!(proposal_type, ProposalType::SELF_REMOVE);
```

Comment:
I don't think external committer can propose  (its own) self-remove for the same reason as above?

Reply:
Ah that's a good point; I've removed it. Thanks.